### PR TITLE
#8 add new article

### PR DESCRIPTION
**Pull request by martuto31**
Don't merge before #30_Articles is merged.

### CODE REVIEW

**Review by MBocharskyi — APPROVED** [review]

File: SportsHub/SportsHub.Api/Controllers/ArticleController.cs [hunk 1]
```diff
@@ -31,5 +32,19 @@ public async Task<ActionResult<ArticleResponseDTO>> GetArticleByTitleAsync(strin
             var articleResponse = _mapper.Map<ArticleResponseDTO>(article);
             return Ok(articleResponse);
         }
+
+        [Authorize(Roles = "Admin")]
+        [HttpPost]
+        public async Task<IActionResult> CreateArticleAsync([FromBody] CreateArticleDTO adminInput)
+        {
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.Api/Controllers/ArticleController.cs:39]: Add validation for the input fields, and check restrictions in the database and in GitHub acceptance. Add a Fluent validator for the input restrictions, and check uniqueness in service layer

File: SportsHub/SportsHub.Api/Controllers/ArticleController.cs [hunk 2]
```diff
@@ -31,5 +43,26 @@ public async Task<ActionResult<ArticleResponseDTO>> GetArticleByTitleAsync(strin
...
+            {
+                var response = _generateModelStateDictionary.modelStateDictionary(validationResult);
+                return ValidationProblem(response);
+            }
+
+            bool createdSuccessful = await _articleService.CreateArticleAsync(adminInput);
+
+            if(createdSuccessful)
+            {
+                return Ok("Article created successfully.");
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.Api/Controllers/ArticleController.cs:62]: move to constants

File: SportsHub/SportsHub.Api/Controllers/ArticleController.cs [hunk 3]
```diff
@@ -31,5 +43,26 @@ public async Task<ActionResult<ArticleResponseDTO>> GetArticleByTitleAsync(strin
             var articleResponse = _mapper.Map<ArticleResponseDTO>(article);
             return Ok(articleResponse);
         }
+
+        [Authorize(Roles = "Admin")]
+        [HttpPost("AddArticle")]
+        public async Task<IActionResult> CreateArticleAsync([FromBody] CreateArticleDTO adminInput)
+        {
+            ValidationResult validationResult = await _articleValidator.ValidateAsync(adminInput);
+            if(!validationResult.IsValid)
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.Api/Controllers/ArticleController.cs:52]: Does not it work without manual calls?

File: SportsHub/SportsHub.Application/Services/ArticleService.cs [hunk 4]
```diff
@@ -21,5 +23,30 @@ public async Task<IEnumerable<Article>> GetAllAsync()
...
+
+            if (articleExists)
+            {
+                return false;
+            }
+
+            var article = new Article()
+            {
+                Title = adminInput.Title,
+                CategoryId = adminInput.CategoryId,
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.Application/Services/ArticleService.cs:39]: add validation if CategoryId exists, please. If not then false

File: SportsHub/SportsHub.Api/Controllers/ArticleController.cs (hunk 3 above)
Comment (NataliiaBidiuk) [reply SportsHub/SportsHub.Api/Controllers/ArticleController.cs:68]: I understand, you can use manual here

File: SportsHub/SportsHub.Application/Authentication/Models/DTOs/CreateArticleDTO.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿namespace SportsHub.AppService.Authentication.Models.DTOs
+{
+    public class CreateArticleDTO
+    {
+        public int ArticleId { get; set; }
```
Comment (NataliiaBidiuk) [inline SportsHub/SportsHub.Application/Authentication/Models/DTOs/CreateArticleDTO.cs:5]: ArticleId should be auto-generated, delete that field

File: SportsHub/SportsHub.Application/Services/ArticleService.cs (hunk 4 above)
Comment (martuto31) [reply SportsHub/SportsHub.Application/Services/ArticleService.cs:46]: Added validation for CategoryId but I am not sure if that's the right approach.

File: SportsHub/SportsHub.Application/Authentication/Models/DTOs/CreateArticleDTO.cs (hunk 5 above)
Comment (martuto31) [reply SportsHub/SportsHub.Application/Authentication/Models/DTOs/CreateArticleDTO.cs]: Deleted.

File: SportsHub/SportsHub.Application/Services/ArticleService.cs (hunk 4 above)
Comment (martuto31) [reply SportsHub/SportsHub.Application/Services/ArticleService.cs:46]: And is it better to move the GetCategoryById method into a Category service?

Comment (NataliiaBidiuk) [reply SportsHub/SportsHub.Application/Services/ArticleService.cs:46]: you can calls Category repository here

**Review by NataliiaBidiuk — APPROVED** [review]
resolve conflicts, then ping me. i will merge it

**Pull request merged**